test/unit/longrunning.ts

```diff
@@ -698,3 +773,512 @@ describe('longrunning', () => {
     });
   });
 });
+
+describe('diregapic longrunning', () => {
```

Comment:
May I ask you to move this to a separate file? 1000+ lines is a lot :)  This will also simplify all the test initialization code since you won't need to do `if (diregapic)` in it.